Taskfile.yml

```diff
@@ -1,6 +1,12 @@
 version: "3"
 
 tasks:
+  setup:
+    - python -m venv .venv # Create a virtual environment
+    - source .venv/bin/activate # Activate the virtual environment
+    - python -m pip install -r requirements.txt
+    - dbt deps
+    - npm --prefix ./reports install
```

Comment:
also, we should add `pre-commit install` to the setup (i need to add information on this to the docs i'm realizing), i realized the `requirements.txt` changes here didn't get alpha-sorted which means pre-commit didn't run, and that would make sense because i don't call it out in the README or setup tasks!